gcloud/drivers.yaml

```diff
@@ -316,6 +318,48 @@ drivers:
 
                      retbytes, err := driver.RPCCallRaw("driver:gcloud-kms", "decrypt", cipherbytes)
 
+      gcloud-secret:
+        name: gcloud-secret
+        type: builtin
+        handlerData:
+          shortName: gcloud-secret
+
+        meta:
+          description:
+            - >
+              This driver enables Honeydipper to fetch items stored in google secret manager.
+
+            - >
+              With access to google secret manager, Honeydipper doesn't have to rely on cipher texts stored directly into the
+              configurations in the repo. Instead, it can query the google secret manager, and get access to the secrets based
+              on the permissions granted to the identity it uses. :code:`DipperCL` uses a keyword interpolation to detect the items
+              that need to be looked up using :code:`LOOKUP[<driver>,<key>]`. See blow for example.
+
+            - example: |
+                mydata: LOOKUP[gcloud-secret,projects/foo/secrets/bar/versions/latest]
+
+            - >
+              As of now, the driver doesn't take any configuration other than the generic `api_timeout`. It uses the default service
+              account as its identity.
+
+          RPCs:
+            - name: lookup
+              description: Lookup a secret in google secret manager
+              parameters:
+                - name: "*"
+                  description: The whole payload is used as a byte array of string for the key
+              returns:
+                - name: "*"
+                  description: The whole payload is a byte array of plaintext
```

Comment:
can / should we enforce the length and content of the string vs. allowing any string (in the driver, not in config-essentials)? I think this would be smart (based on valid secret names; and throw a configcheck error on invalid string).

Reply:
It needs to be done in the go code. Do you mind making it an issue? This is good for beginners to take on, and I can help.

Reply:
yes, that's what I was thinking as well, and yes, I can add an issue.

Reply:
https://github.com/honeydipper/honeydipper/issues/388